### ISSUE TITLE
Could not build arm64 contains xcframework for iOS simulator by Carthage.

### ISSUE THREAD

**Issue by r-plus** [realm/realm-swift#7151]
Carthage built xcframework for realm will not contains arm64 slice for ios simulator, thus I could not build my project for ios simulator on M1 mac.

## Goals
`carthage bootstrap --platform iOS --use-xcframeworks --no-use-binaries realm-cocoa`
command will compile arm64, i386 and x86_64 contains xcframework for ios simulator.

## Expected Results
built xcframework contains `ios-arm64_i386_x86_64-simulator` directory.

## Actual Results
contains `ios-i386_x86_64-simulator`

## Steps for others to Reproduce

```
$ carthage version
0.37.0

$ echo 'github "realm/realm-cocoa"' > Cartfile

$ carthage bootstrap --platform iOS --use-xcframeworks --no-use-binaries realm-cocoa
*** No Cartfile.resolved found, updating dependencies
*** Fetching realm-cocoa
*** Checking out realm-cocoa at "v10.7.0"
*** xcodebuild output can be found in /var/folders/st/df0gjt9x5tl0cv7fw0xg5s9m0000gn/T/carthage-xcodebuild.LjYtzX.log
*** Building scheme "RealmSwift" in Realm.xcworkspace
*** Building scheme "Realm iOS static" in Realm.xcworkspace
*** Building scheme "Realm" in Realm.xcworkspace

$ ls -1 Carthage/Build/Realm.xcframework
Info.plist
ios-arm64_armv7
ios-i386_x86_64-simulator

$ ls -1 Carthage/Build/RealmSwift.xcframework
Info.plist
ios-arm64_armv7
ios-i386_x86_64-simulator
```

## Code Sample
n/a

## Version of Realm and Tooling
Realm framework version: v10.7.0

Realm Object Server version: n/a

Xcode version: 12.4

iOS/OSX version: n/a, using mac is M1 mac 11.2.0

Dependency manager + version: Carthage 0.37.0

**Comment by pavel-ship-it** [comment]
Hi @r-plus,
There is an issue with Carthage and Xcode 12+ - in two words - it can't build arm64 for iOS and M1 (or Simulator in this case) at same time. Please refer to the workaround described in Carthage docs - https://github.com/Carthage/Carthage#building-platform-specific-framework-bundles-default-for-xcode-11-and-below

**Comment by r-plus** [comment]
@pavel-ship-it That issue is already resolved by xcframeworks. Latest carthage already supported to building xcframeworks. 
This issue talk about it.

**Comment by pavel-ship-it** [comment]
@r-plus, you right. My bad. I'll take a deeper look in to this problem.

**Comment by edfanhouse** [comment]
Hello,

I found a workaround with commenting arm64 exclusion on "Base.config" when building with Carthage :

<img width="606" alt="Capture d’écran 2021-03-02 à 09 50 44" src="https://user-images.githubusercontent.com/1589679/109634978-24cad100-7b4a-11eb-8e9c-9991e203cce0.png">

**Comment by r-plus** [comment]
@edfanhouse nice! that it!

#### Remaining issue but maybe this is not blocking
- workflow to create pre-build binary for carthage that uploaded to github releases like [RxSwift](https://github.com/ReactiveX/RxSwift/releases)
AFAIK, `carthage archive` command is not yet support xcframework.
NOTE: carthage 0.37 is not yet support download pre-build xcframework.

**Comment by pavel-ship-it** [comment]
@r-plus , @edfanhouse thanks for help. I'll create a pull request to enable arm64 builds

**Comment by r-plus** [comment]
for link: https://github.com/realm/realm-cocoa/pull/7158

**Comment by pavel-ship-it** [comment]
Fixed in #7158 
Available on master. Will be released in the upcoming version.
